rpctest/rpcharness_test.go

```diff
@@ -972,31 +1064,736 @@ func testSendFrom(r *Harness, t *testing.T) {
+	//var totalAmountToSend dcrutil.Amount
+	totalAmountToSend := dcrutil.Amount(0)
+
+	for i, acct := range accountNames {
+		addr, err := wcl.GetNewAddress(acct)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Set the amounts to send to each address
+		addresses = append(addresses, addr)
+		addressAmounts[addr] = amountsToSend[i]
+		totalAmountToSend += amountsToSend[i]
+	}
+
+	// SendMany to two addresses
+	_, err = wcl.SendMany("default", addressAmounts)
+	if err != nil {
+		t.Fatalf("sendmany failed: %v", err)
+	}
+
+	// This should add 5 results: coinbase send, 2 receives, 2 sends
+	listSentMany, err := wcl.ListTransactionsCount("*", 99999999)
+	if err != nil {
+		t.Fatal(`Listtransactions failed.`)
+	}
+	if len(listSentMany) != len(txListAll)+5 {
+		t.Fatalf("Expected %v tx results, got %v", len(txListAll)+5,
+			len(listSentMany))
+	}
+}
+
+func testGetSetRelayFee(r *Harness, t *testing.T) {
+	// dcrrpcclient does not have a getwalletfee or any direct method, so we
+	// need to use walletinfo to get.  SetTxFee can be used to set.
+
+	// Wallet RPC client
+	wcl := r.WalletRPC
+
+	// Increase the ticket fee so these SSTx get mined first
+	walletInfo, err := wcl.WalletInfo()
+	if err != nil {
+		t.Fatal("WalletInfo failed:", err)
+	}
+	// Save the original fee
+	origTxFee, err := dcrutil.NewAmount(walletInfo.TxFee)
+	if err != nil {
+		t.Fatal("Invalid Amount:", walletInfo.TxFee)
+	}
+	// Increase fee by 50%
+	newTxFeeCoin := walletInfo.TxFee * 1.5
+	newTxFee, _ := dcrutil.NewAmount(newTxFeeCoin)
+	if err != nil {
+		t.Fatal("Invalid Amount:", newTxFeeCoin)
+	}
+
+	err = wcl.SetTxFee(newTxFee)
+	if err != nil {
+		t.Fatal("SetTxFee failed:", err)
+	}
+
+	// Check that wallet thinks the fee is as expected
+	walletInfo, err = wcl.WalletInfo()
+	if err != nil {
+		t.Fatal("WalletInfo failed:", err)
+	}
+	newTxFeeActual, err := dcrutil.NewAmount(walletInfo.TxFee)
+	if err != nil {
+		t.Fatal("Invalid Amount:", walletInfo.TxFee)
+	}
+	if newTxFee != newTxFeeActual {
+		t.Fatalf("Expected tx fee %v, got %v.", newTxFee, newTxFeeActual)
+	}
+
+	// Create a transaction and compute the effective fee
+	accountName := "testGetSetRelayFee"
+	err = wcl.CreateNewAccount(accountName)
+	if err != nil {
+		t.Fatal("Failed to create account.")
+	}
+
+	// Grab a fresh address from the test account
+	addr, err := r.WalletRPC.GetNewAddress(accountName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// SendFromMinConf to addr
+	amountToSend := dcrutil.Amount(700000000)
+	txid, err := wcl.SendFromMinConf("default", addr, amountToSend, 0)
+	if err != nil {
+		t.Fatalf("sendfromminconf failed: %v", err)
+	}
+
+	newBestBlock(r, t)
+	time.Sleep(2 * time.Second)
+	// Give the tx a sensible MsgTx().TxIn[:].ValueIn values
+
+	// Compute the fee
+	rawTx, err := r.Node.GetRawTransaction(txid)
+	if err != nil {
+		t.Fatalf("getrawtransaction failed: %v", err)
+	}
+
+	fee := getWireMsgTxFee(rawTx)
+	feeRate := fee.ToCoin() / float64(rawTx.MsgTx().SerializeSize()) * 1000
```

Comment:
Say this was done in Amount, there would still be division by size, and there might be a remainder, essentially an inaccuracy.